client/components/tinymce/plugins/insert-menu/plugin.jsx

```diff
@@ -26,11 +26,28 @@ const initialize = editor => {
 		cmd: menuItems[ 0 ].cmd,
 		menu: menuItems.map( ( { name } ) => editor.menuItems[ name ] ),
 		onPostRender() {
-			ReactDOM.render(
-				<Gridicon icon="add-outline" />,
-				this.$el[ 0 ].children[ 0 ]
-			);
-		}
+			ReactDOM.render( <Gridicon icon="add-outline" />, this.$el[ 0 ].children[ 0 ] );
```

Comment:
This is causing memory leaks every time the editor is opened and closed. The `Gridicon` is repeatedly mounted, but there is no `unmountComponentAtNode` call. Better use `innerHTML = renderToString( ... )`. See how the menu items are rendered a few lines above.